### PR TITLE
Feature/#625 add priority queue

### LINKED ISSUE #625: Add priority queue
Add the possibility to have workers waiting only on high priority orders, e.g. for paying customers.

### PR DESCRIPTION
**Pull request by hixi**
Reviewed by:
- [x] @das-g 

Closes #625

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -108,6 +108,35 @@ services:
       - osm-world
       - osmboundaries
       - osmaxx-internal
+  worker-high:
```
Comment (das-g) [inline docker-compose.yml]: We planned to name this `worker-exclusive` IIRC.

File: docker-compose.yml [hunk 2]
```diff
@@ -108,6 +108,35 @@ services:
...
+      # only listens for jobs in this specific queue
+      - WORKER_QUEUES=high
+    depends_on:
+      - conversionserviceredis
+      - world-database
+      - osmboundaries-database
+    networks:
+      - osm-world
+      - osmboundaries
+      - osmaxx-internal
```
Comment (das-g) [inline docker-compose.yml]: Can the duplication between `worker` and `worker-exclusive` (currently `worker-high`) configurations be reduced?

File: osmaxx_conversion_service/config/settings/common.py [hunk 3]
```diff
@@ -252,6 +252,13 @@
         'PASSWORD': '',
         'DEFAULT_TIMEOUT': int(timedelta(days=1).total_seconds()),
     },
+    'high': {
+        'HOST': env.str('REDIS_HOST', default='localhost'),
+        'PORT': env.str('REDIS_PORT', default=6379),
+        'DB': 0,
+        'PASSWORD': '',
+        'DEFAULT_TIMEOUT': int(timedelta(days=1).total_seconds()),
+    },
```
Comment (das-g) [inline osmaxx_conversion_service/config/settings/common.py]: Consider eliminating the duplication here by pulling the identical dicts out to a local (private?) variable.

File: docker-compose.yml (hunk 2 above)
Comment (hixi) [inline docker-compose.yml]: somewhat:

```yml
  worker-base:
    image: geometalab/osmaxx-worker:${DEPLOY_VERSION}
    extends:
      file: docker-compose-common.yml
      service: conversionbase
    command: [honcho, -f, ./osmaxx_conversion_service/Procfile.worker, start]
    volumes:
      - osm_data:/var/data/osm-planet
      - worker-data:/data/media/job_result_files
    environment:
      - REDIS_HOST=redis
      - REDIS_PORT=6379
      - DJANGO_SETTINGS_MODULE=osmaxx_conversion_service.config.settings.worker
  worker:
    extends:
      file: docker-compose.yml
      service: worker-base
    volumes-from:
      - worker-base
    links:
      - conversionserviceredis:redis
      - world-database
      - osmboundaries-database
    environment:
      - REDIS_HOST=redis
      - REDIS_PORT=6379
      - DJANGO_SETTINGS_MODULE=osmaxx_conversion_service.config.settings.worker
    depends_on:
      - conversionserviceredis
      - world-database
      - osmboundaries-database
    networks:
      - osm-world
      - osmboundaries
      - osmaxx-internal
  worker-high:
    image: geometalab/osmaxx-worker:${DEPLOY_VERSION}
    extends:
      file: docker-compose.yml
      service: worker-base
    command: [honcho, -f, ./osmaxx_conversion_service/Procfile.worker, start]
    volumes:
      - osm_data:/var/data/osm-planet
      - worker-data:/data/media/job_result_files
    links:
      - conversionserviceredis:redis
      - world-database
      - osmboundaries-database
    environment:
      # only listens for jobs in this specific queue
      - WORKER_QUEUES=high
    depends_on:
      - conversionserviceredis
      - world-database
      - osmboundaries-database
    networks:
      - osm-world
      - osmboundaries
      - osmaxx-internal
```

In addition of adding more lines (longer than duplication), it also adds the issue of having a non-functioning worker-base, which always should be remembered to not be started.

**Comment by hixi** [comment]
@das-g I changed some things, discovered a bug in the calculation for the estimation, which I corrected on here. Can you re-review, please?

**Pull request merged**